nunit/MVP-Advance/SeleniumNUnit/Pages/ManageRequests.cs

```diff
@@ -0,0 +1,288 @@
+        {
+            //Populate Excel and get request name
+            ExcelLib.PopulateInCollection(Base.ExcelPath, "ManageRequests");
+            string skill = ExcelLib.ReadData(2, "Title");
+
+            //Click "Sent Requests"
+            ClickSentRequests();
+
+            //Look for request and return request text
+            for (int i = 0; i < assertSentTitles.Count(); i++)
+            {
+                if (assertSentTitles[i].Text.Equals(skill))
+                {
+                    return assertSentTitles[i].Text;
+                }
+                else
+                {
+                    return "No matching request found.";
+                }
+            }
+            return "There's no request.";
+        }
+
+        public void WithdrawRequest()
+        {
+            //Populate Excel and get request name
+            ExcelLib.PopulateInCollection(Base.ExcelPath, "ManageRequests");
+            string skill = ExcelLib.ReadData(2, "Title");
+
+            //Click sent requests
+            ClickSentRequests();
+
+            //Get skill index in Sent Requests table
+            string index = "//div[@id='sent-request-section']//tbody/tr[" + GetSentSkillIndex(skill) + "]/td[8]/button";
+
+            //Click button Withdraw of correspondent skill
+            IWebElement btnWithdraw = driver.FindElement(By.XPath(index));
+            btnWithdraw.Click();
+        }
+
+        public void DeclineRequest()
+        {
+            //Populate Excel and get request name
+            ExcelLib.PopulateInCollection(Base.ExcelPath, "ManageRequests");
+            string skill = ExcelLib.ReadData(2, "Title");
+
+            //Log in the Seller account
+            SwitchAccount(2);
+
+            //Click Received Request
+            ClickReceivedRequests();
+
+            //wait for table to be displayed
+            WaitHelpers.WaitToBeVisible(driver, "XPath", categoryColumn, 10);
+
+            //Get skill index in Received Requests table
+            string index = "//div[@id='received-request-section']//tbody/tr[" + GetReceivedSkillIndex(skill) + "]/td[8]/button[2]";
+
+            //Click button Decline
+            IWebElement btnDelecine= driver.FindElement(By.XPath(index));
+            btnDelecine.Click();
+        }
+
+        public void AcceptRequest()
+        {
+            //Populate Excel and get request name
+            ExcelLib.PopulateInCollection(Base.ExcelPath, "ManageRequests");
+            string skill = ExcelLib.ReadData(2, "Title");
+
+            //Log in the Seller account
+            SwitchAccount(2);
+
+            //Click Received Request
+            ClickReceivedRequests();
+
+            //Wait for table to be displayed
+            WaitHelpers.WaitToBeVisible(driver, "XPath", categoryColumn, 10);
+
+            //Get skill index in Sent Requests table
+            string index = "//div[@id='received-request-section']//tbody/tr[" + GetReceivedSkillIndex(skill) + "]/td[8]/button[1]";
```

Comment:
any reason this is not following the page object style?

Reply:
pretty much where ever you have done this please

Reply:
Same as it's a dynamic XPath